test/unit/math/opencl/device_functions/log1p_exp_test.cpp

```diff
@@ -0,0 +1,33 @@
+#ifdef STAN_OPENCL
+#include <stan/math.hpp>
+#include <stan/math/opencl/device_functions/log1p_exp.hpp>
+#include <stan/math/opencl/kernel_cl.hpp>
+#include <gtest/gtest.h>
+#include <test/unit/math/expect_near_rel.hpp>
+#include <string>
+
+static const std::string test_kernel_code
+    = STRINGIFY(__kernel void test(__global double *B, __global double *A) {
+        const int i = get_global_id(0);
+        B[i] = log1p_exp(A[i]);
+      });
+
+const stan::math::opencl_kernels::kernel_cl<
+    stan::math::opencl_kernels::out_buffer,
+    stan::math::opencl_kernels::in_buffer>
+    log1p_exp("test", {stan::math::opencl_kernels::log1p_exp_device_function,
+                       test_kernel_code});
+
+TEST(MathMatrixCL, log1p_exp) {
+  Eigen::VectorXd a = Eigen::VectorXd::Random(1000).array() * 0.9999;
+
+  stan::math::matrix_cl<double> a_cl(a);
+  stan::math::matrix_cl<double> res_cl(1000, 1);
+  log1p_exp(cl::NDRange(1000), res_cl, a_cl);
+  Eigen::VectorXd res = stan::math::from_matrix_cl<-1, 1>(res_cl);
+
+  stan::test::expect_near_rel("log1p_exp (OpenCL)", res,
```

Comment:
Edge cases need to be covered: https://github.com/stan-dev/math/blob/502487c511594ccb93eb979df5b8fe163becb417/test/unit/math/prim/scal/fun/log1p_exp_test.cpp